tests/components/calendar/test_demo.py

```diff
@@ -0,0 +1,24 @@
+"""The tests for the demo calendar component."""
+from datetime import timedelta
+
+from homeassistant.bootstrap import async_setup_component
+import homeassistant.util.dt as dt_util
+
+
+async def test_api_calendar_demo_view(hass, aiohttp_client):
+    """Test the calendar demo view."""
+    await async_setup_component(hass, 'calendar',
+                                {'calendar': {'platform': 'demo'}})
+    client = await aiohttp_client(hass.http.app)
+    response = await client.get(
+        '/api/calendar/future_event')
+    assert response.status == 400
+    start = dt_util.now()
+    end = start + timedelta(days=1)
+    response = await client.get(
+        '/api/calendar/future_event?start={}&end={}'.format(start.isoformat(),
```

Comment:
Wait, I am very confused now. How come we need to reference to calendars by … event name ("future event") ?

Reply:
this is the device_id `calendar.future_event` without the `calendar.`

Reply:
Can we rename the demo component calendars to make more sense ? Something like calendar_1, calendar_2